File: Part 1/Programming Assignment 6/solution.py
with open("algo1-programming_prob-2sum.txt", "r") as f:
    lines = f.readlines()
    nums = []
    for line in lines:
        nums.append(int(line))
    nums = list(set(nums))
print("finished reading file")
seen = set()
freqdict = {}
for n in nums:
    if n in freqdict:
        freqdict[n] += 1
    else:
        freqdict[n] = 1

count = 0
for t in range(-10000, 10000):
    print(t)
    for s in freqdict:
        x = t-s
        # if x == s:
        #     if freqdict[x] >= 2:
        #         count += 1
        #         break
        # else:
        if x !=s and x in freqdict:
            count += 1
            break
print(count)
# for i in range(len(nums)):
#     for j in range(i+1, len(nums)):
#         total = nums[i] + nums[j]
#         if -10000 <=total <= 10000:
#             seen.add(total)

# print(len(seen))

File: Part 1/Programming Assignment 6/solution.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

int main(){
    ifstream file("algo1-programming_prob-2sum.txt");
    // file.open();
    //file.open("")
    string line;
    vector<int> nums;
    long int x;
    while(getline(file, line)){
        //istringstream strstream(line);
        nums.push_back(stoi(line));
        //cout << x;
    }
    unordered_map<int, int> freqdict;
    for(const int &n:nums){
        if(freqdict.find(n)==freqdict.end()){
            freqdict[n] = 0;
        }
        freqdict[n] += 1;
    }
    cout <<  freqdict.size();
    string fuck;
    cin >> fuck;
    int count = 0;
    for(int t=-10000; t <= 10000; t++){
        cout << t << endl;
        for(auto it=freqdict.begin(); it!=freqdict.end(); ++it){
            
            x = t - it->first;
            if(x==it->first){
                if(it->second >= 2){
                    count ++;
                    break;
                }
            }else{
                if(freqdict.find(x)!=freqdict.end()){
                    count ++;
                    break;
                }
            }
        }
    }
    cout << "Number of fricking integers: " << count;
}